src/main/java/org/veupathdb/service/eda/ss/model/filter/DateRangeFilter.java

```diff
@@ -28,8 +29,9 @@ public String getFilteringAndClausesSql() {
   }
 
   @Override
-  public Predicate<LocalDateTime> getPredicate() {
-    return date -> !date.isBefore(_min) && !date.isAfter(_max);
+  public Predicate<Long> getPredicate() {
+    return date -> date >= _min.toInstant(ZoneOffset.UTC).toEpochMilli()
```

Comment:
seems we could do the conversion of min and max in the constructor instead of each call to the predicate

Reply:
Good point, fixed